src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py

```diff
@@ -367,8 +367,9 @@ def load_arguments(self, _):
             c.argument('use_unmanaged_disk', action='store_true', help='Do not use managed disk to persist VM')
             c.argument('data_disk_sizes_gb', nargs='+', type=int, help='space-separated empty managed data disk sizes in GB to create')
             c.ignore('image_data_disks', 'storage_account_type', 'public_ip_type', 'nsg_type', 'nic_type', 'vnet_type', 'load_balancer_type', 'app_gateway_type')
-            c.argument('os_caching', options_list=['--storage-caching', '--os-disk-caching'], help='Storage caching type for the VM OS disk.', arg_type=get_enum_type([CachingTypes.read_only.value, CachingTypes.read_write.value], default='ReadWrite'))
-            c.argument('data_caching', options_list=['--data-disk-caching'], help='Storage caching type for the VM data disk(s).', arg_type=get_enum_type(CachingTypes))
+            c.argument('os_caching', options_list=['--storage-caching', '--os-disk-caching'], help='Storage caching type for the VM OS disk. Default: ReadWrite', arg_type=get_enum_type(CachingTypes))
```

Comment:
Did this change?  Previously we couldn't just use an enum on caching type because it allowed "None" which compute said was invalid.

Reply:
I actually confirmed with @gatneil, that it would not make sense to block None as it is supported by service, but VMSS does need to address a server bug that OS disk should default to ReadWrite, which I am tracking by #5736

Reply:
Yeah, "None" is a valid value. We don't necessarily recommend it unless you know what you're doing, but it's a valid value :)